src/nodes/MemberAccess.js

```diff
@@ -4,15 +4,42 @@ const {
   }
 } = require('prettier/standalone');
 
-const isBeginnigOfChain = (path) => {
-  const parentNodeType = path.getParentNode().type;
+const isEndOfChain = (node, path) => {
+  let i = 0;
+  let currentNode = node;
+  let parentNode = path.getParentNode(i);
+  while (parentNode) {
+    // If direct ParentNode is a MemberAcces we are not at the end of the chain
+    if (parentNode.type === 'MemberAccess') return false;
 
-  if (parentNodeType === 'MemberAccess') return false;
-  if (parentNodeType === 'FunctionCall') {
-    const grandParentNodeType = path.getParentNode(1).type;
-    return grandParentNodeType !== 'MemberAccess';
-  }
+    // If direct ParentNode is a FunctionCall and currentNode is not the expression
+    // then it must be and argument in which case it is the end of the chain.
+    if (
+      parentNode.type === 'FunctionCall' &&
+      currentNode !== parentNode.expression
+    )
+      break;
+
+    // If direct ParentNode is an IndexAccess and currentNode is not the base
+    // then it must be the index in which case it is the end of the chain.
+    if (parentNode.type === 'IndexAccess' && currentNode !== parentNode.base)
+      break;
 
+    if (
+      parentNode.type === 'BinaryOperation' ||
+      parentNode.type === 'UnaryOperation' ||
+      parentNode.type === 'IfStatement' ||
+      parentNode.type === 'WhileStatement' ||
+      parentNode.type === 'ForStatement' ||
+      parentNode.type === 'VariableDeclarationStatement' ||
+      parentNode.type === 'ExpressionStatement'
```

Comment:
can we alpha order these?

Reply:
This list of nodes feel a bit ad hoc, what's the reason behind them?

Btw, one way to both simplify this code _and_ document its intention a bit better is to do something like this:

```js
const nodesWhereWeBreakForSomeReason = ['BinaryOperation', 'UnaryOperation', ...]
if (nodesWhereWeBreakForSomeReason.includes(parentNode.type)) {
  break
}
```

where `nodesWhereWeBreakForSomeReason` has a proper name :smile: